tests/integration/tasks/test_databricks.py

```diff
@@ -305,35 +305,40 @@ def test_upsert():
             "ContactEMail": "Alexander.Chambers@KentLyons.com",
             "MailingCity": "New Nicholaschester",
         },
+        {
+            "Id": "UpsertTest2",
+            "AccountId": "Updated!EHNYKjSZsiy",
+            "Name": "new upsert-2",
+            "FirstName": "Updated",
+            "LastName": "Carter2",
+            "ContactEMail": "Adam.Carter@TurnerBlack.com",
+            "MailingCity": "Updated!Jamesport",
+        },
     ]
-    upsert_data = [
+
+    data = [
         {
-            "Id": "wRACnHTeuw",
-            "AccountId": "Upserted!",
-            "Name": "Upsert-Merritt",
-            "FirstName": "Melody",
-            "LastName": "Cook",
-            "ContactEMail": "Melody.Cook@ScottMerritt.com",
-            "MailingCity": "Elizabethfurt",
+            "Id": "UpsertTest2",
+            "AccountId": "Updated!EHNYKjSZsiy",
+            "Name": "new upsert-2",
+            "FirstName": "Updated",
+            "LastName": "Carter2",
+            "ContactEMail": "Adam.Carter@TurnerBlack.com",
+            "MailingCity": "Updated!Jamesport",
         }
     ]
+
     table_name = "raw.test_table"
 
-    match = {
-        "Id": "main.Id",
-        "AccountId": "updates.AccountId",
-        "Name": "updates.Name",
-        "FirstName": "updates.FirstName",
-        "LastName": "updates.LastName",
-        "ContactEMail": "updates.ContactEMail",
-        "MailingCity": "main.MailingCity",
-    }
+    match = "Id= main.Id, AccountId= updates.AccountId, Name= updates.Name, FirstName= updates.FirstName, LastName= updates.LastName, ContactEMail= updates.ContactEMail, MailingCity= main.MailingCity"
+    data = pd.DataFrame(data)
+    query = databricks.build_merge_query(table_name, "Id", data, match)
 
     df = pd.DataFrame(source_data)
     df = df.sort_values(by="Id")
     df = df.reset_index(drop=True)
 
-    databricks.upsert(table_name, upsert_data, match, "Id")
+    databricks.query(query)
```

Comment:
1. `databricks.upsert()` was using `databricks.query()` before to run the upsert statement internally; the the difference is now the query building and execution are split up.

2. `databricks.build_merge_query()` builds the upsert query only; it does not run it. We need to run the query/statement somehow. If the name `query` is confusing, I can rename the `query` function to `sql`; this way it would be clearer that it's for general SQL statements, both querying data and performing operations (ex. create table/schema, delete table etc.)